syntax = "proto3";
package fileutil;

service CompressService
{
	rpc s_compress (stream CompressReq) returns (stream CompressRes) {}
}

message CompressReq
{
	bytes source = 1;
}

message CompressRes
{
	bytes compressed = 1;
}

service UncompressService
{
	rpc s_uncompress (stream UncompressReq) returns (stream UncompressRes) {}
}

message UncompressReq
{
	bytes compressed = 1;
}

message UncompressRes
{
	bytes uncompressed = 1;
}

service EncodeService
{
	rpc s_encode (stream EncodeReq) returns (stream EncodeRes) {}
}

message EncodeReq
{
	byte source = 1;
}

message EncodeRes
{
	byte encoded = 1;
}

service DecodeService
{
	rpc s_decode (stream DecodeReq) returns (stream DecodeRes) {}
}

message DecodeReq
{
	byte encoded = 1;
}

message DecodeRes
{
	byte decoded = 1;
}